app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts

```diff
@@ -536,20 +545,22 @@ export class ParsedCommand {
 export class AppCommandParser {
     private store: Store;
     private channelID: string;
+    private teamID: string;
     private rootPostID?: string;
     private intl: Intl;
 
     forms: {[location: string]: AppForm} = {};
 
-    constructor(store: Store|null, intl: Intl, channelID: string, rootPostID = '') {
+    constructor(store: Store|null, intl: Intl, channelID: string, teamID = '', rootPostID = '') {
```

Comment:
Typescript kind of failed us here because `teamID` was added in the middle of the function signature and made optional, even though it is required for all of the instances of creating this object. This created bugs that needed to be caught during review, while compile-time checks did not catch it.

Should we have `teamID` be required in all of the functions that accept it in `AppCommandParser`? Also, should `getAppContext` use `this.teamID` instead of `getCurrentTeamId()`?

Reply:
Regarding making it necessary, would not solve the problem, and I am not sure whether it is completely necessary.

Regarding the context, yes, it will be better to set it to team ID specified on construction.